### DISCUSSION TITLE
Using Auto CLI as a library

### DISCUSSION THREAD

**Discussion by laughedelic** [intuit/auto#1964] [Q&A] [answered]
I'm trying to do something like this:

```typescript
import { parseArgs, runCli } from "auto";

const [command, args] = parseArgs();
runCli(command, args);
```

Based on the code I see [here](https://github.com/intuit/auto/blob/main/packages/cli/src/bin/auto.ts). But of course, these imports don't work. If I understand correctly, the only public export from that module is this

https://github.com/intuit/auto/blob/3c516648626fc0a93395f8ec00b2afd342c520b1/packages/cli/src/index.ts#L1

---

Is it possible to plug `auto` as a library and take advantage of its CLI code? If it's a matter of the module exports, would you be open to adding those functions to the module's public API?

**Comment by hipstersmoothie** [comment, marked as answer]
Yeah you can export the necessary code form there 👍🏼
